entry_test.go

```diff
@@ -43,17 +50,234 @@ func TestEntryPanicln(t *testing.T) {
 		case *Entry:
 			assert.Equal(t, "kaboom", pVal.Message)
 			assert.Equal(t, errBoom, pVal.Data["err"])
+			assert.Regexp(t, `^level=panic msg=kaboom err="boom time"`, outBuffer.String())
 		default:
 			t.Fatalf("want type *Entry, got %T: %#v", pVal, pVal)
 		}
 	}()
 
-	logger := New()
-	logger.Out = &bytes.Buffer{}
+	logger.Out = &outBuffer
 	entry := NewEntry(logger)
 	entry.WithField("err", errBoom).Panicln("kaboom")
 }
 
+// TestEntryFormattingError tests proper handling of any field formatting errors
+func TestEntryFormattingError(t *testing.T) {
+	logger := New()
+	// Turning off timestamp and color marks parsing the result easier:
+	logger.Formatter = &TextFormatter{
+		DisableTimestamp: true,
+		DisableColors:    true,
+	}
+	// Regular Expressions used later
+	keyValPartsPattern := regexp.MustCompile(`([^\s"=]*|"(?:[^\\"]|\\.)*")=([^\s"=]*|"(?:[^\\"]|\\.)*")(?:\s|$)`)
+	commaSepPattern := regexp.MustCompile(", ")
+	quotedCommaSepPattern := regexp.MustCompile(`", "`)
+	// This allows us to have multiple distinct tests, each with a new entry.
+	// While it is really a good idea to break tests out into separate
+	// independent units, we covered all the obvious cases with one multi-part
+	// test. (Multi-part tests share an entry and test the entry state after
+	// each change.)
+	for _, independantTestCase := range [][]struct {
```

Comment:
If there is a single element in the array, we can just remove a slice and a loop level, it will make the unit test easier to maintain.

Reply:
Actually I put this loop here intentionally, to make independant test cases easier to add. What is there now is a single independant test, with multiple phases. This was due to my laziness. I’d rather break this into multiple independent tests then remove this outer loop. Let me see what I can do in that regard.

Reply:
Added more independant test cases to the outer loop. It no longer has just one item.